### DISCUSSION TITLE
Heads up: First time outside contributions require maintainer approval to run Actions

### DISCUSSION THREAD

**Discussion by HonkingGoose** [renovatebot/renovate#9865] [Suggest an Idea]
I notice some first-time PRs are still awaiting a full GitHub Actions test run. This is because GitHub has changed the process to deal with first-time contributors.

TL:DR: To prevent users abusing Action runs to mine crypocurrencies or do other bad stuff, first time contributors now need to have their Action run approved by a maintainer with **write** level access to the repository.

This blogpost explains GitHub's reasons for making the change [GitHub Blog, GitHub Actions: Helping maintainers combat bad actors ](https://github.blog/2021-04-22-github-actions-update-helping-maintainers-combat-bad-actors/)

[GitHub docs, How to approve workflow run from public forks](https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks).

**Comment by viceice** [comment]
@HonkingGoose You can request review from maintainer to let them check and approve the workflow 🙃

**Reply by HonkingGoose** [reply]
> You can request review from maintainer to let them check and approve the workflow

Will do! 😄

**Reply by kirhov** [reply]
> @HonkingGoose You can request review from maintainer to let them check and approve the workflow 🙃

What to do to request review from maintainer?

**Reply by HonkingGoose** [reply]
Take a look at: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/about-pull-request-reviews#re-requesting-a-review

Edit: this only explains how to re-request a review, read my message below to learn about requesting a review when nobody has yet reviewed.

**Reply by HonkingGoose** [reply]
> What to do to request review from maintainer?

You cannot request a review from the maintainers yourself, quote from the GitHub Docs [^docs]:

> **Note:** Pull request authors can't request reviews unless they are either a repository owner or collaborator with write access to the repository.

More information on who can and cannot request a review:

> Owners and collaborators on a repository owned by a user account can assign pull request reviews. Organization members with triage permissions to a repository can assign a pull request review.

You can however always re-request a review from anybody who has already reviewed your PR.

[^docs]: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review

**Reply by kirhov** [reply]
As a First time Contributor what can I do?

**Reply by kirhov** [reply]
Just Wait?

**Reply by HonkingGoose** [reply]
You need to wait until the maintainers review the PR on their own time. 😉

**Reply by kirhov** [reply]
Thanks!

**Reply by viceice** [reply]
First maintainer review will normally done in a few hours after PR creation. 🤗

We probably ignore drafts they are marked as ready or some feedback comment asks for a review. 😉

**Reply by rarkins** [reply]
FYI every time a first time contributor commits to the branch, it needs fresh approval

**Reply by kirhov** [reply]
<img width="679" alt="Capture" src="https://user-images.githubusercontent.com/94075961/141696123-e40414f6-c594-4533-b7e4-babd50f83ee6.PNG">
<img width="577" alt="Capture2" src="https://user-images.githubusercontent.com/94075961/141696129-6bfda81a-4274-4db6-951e-d0ddb6bec4ae.PNG">
I am wondering if the Two Weeks mentioned is really about the Maintainers..

**Reply by rarkins** [reply]
That doesn't look like a message from this repo

**Reply by kirhov** [reply]
its not from this repository.
